test/toc-check.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+require('colors');
+var toc = require('markdown-toc');
+var Diff = require('diff');
+var fs = require("fs");
+var glob = require("glob");
+
+var files = glob.sync("**/*.md", {ignore: ["node_modules/**/*.md"]})
+
+function removeTocHeader(str, ele, arr) {
+  return str.indexOf('Table of Contents') === -1;
+}
+
+function getCurrentToc(str, remove){
+  if(remove === undefined) remove = 0;
+  var startIndex = str.indexOf('Table of Contents') + 19;
+  var endIndex = str.indexOf("## ", startIndex);
+  var result = str.substring(startIndex, endIndex - remove);
+  
+  if(result[result.length -1] == '\n' || result[result.length -1] == ' '){
+    return getCurrentToc(str, remove+1);
+  }else{
+    return result;
+  }
+}
+
+files.forEach(function(file) {
+  var markdown = fs.readFileSync(file).toString();
+  console.log("Reading: " + file);
+
+  if(markdown.includes('Table of Contents')){
+    var generatedToc = toc(markdown, {filter: removeTocHeader, firsth1: false, bullets: "*"}).content;
+    var currentToc = getCurrentToc(markdown);
+
+    if(currentToc !== generatedToc) {
+      var diff = Diff.diffLines(currentToc, generatedToc);
+      diff.forEach(function(part){
+        var color = part.added ? 'green' :
+          part.removed ? 'red' : 'grey';
+        process.stderr.write(part.value[color]);
+      });
+
+      console.log('\n');
```

Comment:
If you're writing the diff to stderr, the newline should be written to stderr as well.